File: sentimental-credit/credit.py
from cs50 import *
import sys

num = get_int("Number: ")
tmp = num
odd = 0
even = 0
x = 0
count = 0

while tmp > 0:
    tmp //= 10
    digit = (tmp % 10) * 2
    odd += (digit // 10) + (digit % 10)
    tmp //= 10

tmp = num
even = tmp % 10

while True:
    tmp = tmp // 100
    even = even + (tmp % 10)
    if tmp <= 0:
        break

if not ((odd + even) % 10 == 0):
    print("INVALID")
    sys.exit()

tmp = num

while not (tmp <= 0):
    count += 1
    tmp = tmp // 10

if count == 15:
    tmp = num
    while tmp >= 100:
        tmp = tmp // 10
    if tmp == 34 or tmp == 37:
        print("AMEX")
        sys.exit()

if count == 16:
    tmp = num
    while tmp >= 100:
        tmp = tmp // 10
    if 51 <= tmp <= 55:
        print("MASTERCARD")
        sys.exit()

if count == 13 or count == 16:
    tmp = num
    while tmp >= 10:
        tmp = tmp // 10
    if tmp == 4:
        print("VISA")
        sys.exit()
print("INVALID")
sys.exit()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long num = get_long("Number: ");
    long tmp = num;
    long odd = 0;
    long even = 0;
    int x;
    int count = 0;

    tmp = num;
    while (tmp > 0)
    {
        tmp /= 10;
        int digit = (tmp % 10) * 2;
        odd += (digit / 10) + (digit % 10);
        tmp /= 10;
    }

    tmp = num;
    even = tmp % 10;
    while (true)
    {
        tmp = tmp / 100;
        even = even + (tmp % 10);
        if (tmp <= 0)
            break;
    }

    if (!((odd + even) % 10 == 0))
    {
        printf("INVALID\n");
        return 0;
    }

    tmp = num;
    do
    {
        count++;
        tmp = tmp / 10;
    }
    while (!(tmp <= 0));

    if (count == 15) // AMERICAN EXPRESS
    {
        tmp = num;
        while (tmp >= 100)
        {
            tmp = tmp / 10;
        }
        if (tmp == 34 || tmp == 37)
        {
            printf("AMEX\n");
            return 0;
        }
    }
    if (count == 16) // MASTER CARD
    {
        tmp = num;
        while (tmp >= 100)
        {
            tmp = tmp / 10;
        }
        if (tmp == 51 || tmp == 52 || tmp == 53 || tmp == 54 || tmp == 55)
        {
            printf("MASTERCARD\n");
            return 0;
        }
    }
    if (count == 13 || count == 16) // VISA
    {
        tmp = num;
        do
        {
            tmp = tmp / 10;
        }
        while (tmp >= 10);

        if (tmp == 4)
        {
            printf("VISA\n");
            return 0;
        }
    }
    printf("INVALID\n");
    return 0;
}